source/history.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+var noop = function () {
+    return false;
+};
+var pushState = window.history.pushState;
```

Comment:
I’d better place it right before line 21 so that the lifetime of the variable is minimal.